Client/game_sa/CStreamingSA.cpp

```diff
@@ -51,13 +49,179 @@ namespace
             memoryUsed = *(DWORD*)VAR_CStreaming_memoryUsed;
         }
     };
-}            // namespace
+
+    constexpr size_t MAX_IMAGES_NUM = 0xFF;
+    constexpr size_t MAX_STREAMS_NUM = 0xFF;
+    constexpr size_t STREAM_HANDLES_FACTOR = 4;
+} // namespace
 
 bool IsUpgradeModelId(DWORD dwModelID)
 {
     return dwModelID >= 1000 && dwModelID <= 1193;
 }
 
+CStreamingSA::CStreamingSA()
+{
+    SetArchivesNum(VAR_DefaultMaxArchives);
+
+    // Copy the default data
+    HANDLE (&defaultStreamingHandlers)[32] = *(HANDLE(*)[32])0x8E4010;
+    SStreamName (&defaultStreamingNames)[32] = *(SStreamName(*)[32])0x8E4098;
+    CArchiveInfo (&defaultAchiveInfo)[8] = *(CArchiveInfo(*)[8])0x8E48D8;
+
+    assert(m_StreamHandles.size() == VAR_DefaultStreamHandlersMaxCount);
+    assert(m_StreamNames.size() == VAR_DefaultStreamHandlersMaxCount);
+    assert(m_Imgs.size() == VAR_DefaultMaxArchives);
+
+    std::memcpy(m_StreamHandles.data(), defaultStreamingHandlers, sizeof(HANDLE) * VAR_DefaultStreamHandlersMaxCount);
+    std::memcpy(m_StreamNames.data(), defaultStreamingNames, sizeof(SStreamName) * VAR_DefaultStreamHandlersMaxCount);
+    std::memcpy(m_Imgs.data(), defaultAchiveInfo, sizeof(CArchiveInfo) * VAR_DefaultMaxArchives);
+}
+
+void CStreamingSA::SetArchivesNum(size_t imagesNum)
+{
+    if (imagesNum > MAX_IMAGES_NUM)
+        return;    
+
+    /*
+        IMGs
```

Comment:
If we use SOLID principles, this should be in different class (S)

Reply:
What exactly should be in a different class? Method SetArchivesNum? Are you sure that a dedicated class make it better? SOLID is awesome but it is not something that should be applied everywhere.

Reply:
I think we're way past the point where any kind of principles can be applied to this codebase 😂